sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -537,7 +537,6 @@ class CodegenContext {
       val funcCode: String =
         s"""
           public int $compareFunc(InternalRow a, InternalRow b) {
-            InternalRow i = null;
```

Comment:
I think `GenerateOrdering.genComparisons()` is expected to generate a PART of comparison method. If the declaration of the variable `i` is in the part, we can't use `GenerateOrdering.genComparisons()` twice or more for the same method.
Declaration of unused variable will not be a problem.